pose_diff/core/pose_diffing.py

```diff
@@ -200,70 +198,68 @@ def frame_increasing(trainer,user,way): #make bigger the fewer frame
         pass
     if exercise == 1:
-        print("Exercise type is shoulder_press")
+        print("Exercise type is pull_up")
         pass
     if exercise == 2:
-        i = 0
-        while True:
-            if i > len(user) - 1:
-                break
-            for idx, part in enumerate(Parts[exercise]):
-                if part >= 0.5:
-                    if (trainer_angle[i][idx] == None or user_angle[i][idx] == None):
-                        continue
-                    if setting_relative_error(trainer_angle[i][idx],user_angle[i][idx]) < margin:
-                        angle_np[i][idx][2] = 1
-                    else:
-                        angle_np[i][idx][2] = 2
+        for pidx, angle in enumerate(angle_np):
+            sum_gap = 0
+            for idx, part in enumerate(AnglePart):
+                if (trainer_angle[pidx][idx] == None or user_angle[pidx][idx] == None):
+                    continue
+                sum_gap = sum_gap + setting_relative_error(trainer_angle[pidx][idx],user_angle[pidx][idx])
+                if setting_relative_error(trainer_angle[pidx][idx],user_angle[pidx][idx]) > margin:
+                    angle[idx][2] = 1
                 else:
-                    angle_np[i][idx][2] = 0
-            i = i+1
-    return angle_np
+                    angle[idx][2] = 2
+            else:
+                angle[idx][2] = 0
+            gap.append(sum_gap)
+    return gap, angle_np
 
 def point_difference(trainer, user, exercise):
     point_np = np.copy(user)
-    margin = 10
+    margin = 4
     gap = []
+    parts= [[] for i in range(18)]
     # Error in RED = 1, Success in GREEN = 2
     if exercise == 0:
         print("Exercise type is Walk")
         pass
     if exercise == 1:
-        print("Exercise type is shoulder_press")
+        print("Exercise type is pull_up")
         pass
     if exercise == 2:
-        i = 0
-        while True:
+        for pidx, point in enumerate(point_np): #Frame
             sum_gap = 0
-            if i > len(user) - 1:
-                break
-            for idx, part in enumerate(Parts[exercise]):
-                if part >= 0.5:
-                    sum_gap = sum_gap + setting_relative_error(trainer[i][idx][0],user[i][idx][0])
-                    if setting_relative_error(trainer[i][idx][0],user[i][idx][0]) > margin:
-                        point_np[i][idx][2] = 1
+            for idx, part in enumerate(Parts[exercise]): #Body_parts
+                if part == 0.5:
+                    parts[idx].append((round(trainer[pidx][idx][0]-user[pidx][idx][0],2),round(trainer[pidx][idx][1] - user[pidx][idx][1],2)))
+                    sum_gap = sum_gap + setting_relative_error(trainer[pidx][idx][0],user[pidx][idx][0])
+                    if (setting_relative_error(trainer[pidx][idx][0],user[pidx][idx][0])\
+                        + setting_relative_error(trainer[pidx][idx][1],user[pidx][idx][1]))/2> margin:
+                        point[idx][2] = 1
```

Comment:
Leave any doc string for this part so that we can understand how comparing function is changed.
Others are good! Good Work, bro

Reply:
Add some doc string. Please check.

Reply:
Thank you